### PR TITLE
Update all Xiph repositories to gitlab.xiph.org

### PR DESCRIPTION
**Pull request by EricFromCanada**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Updates all Xiph projects to use their new [GitLab server](https://gitlab.xiph.org/xiph).

### CODE REVIEW

File: Formula/reminiscence.rb [hunk 1]
```diff
@@ -22,8 +22,8 @@ class Reminiscence < Formula
   uses_from_macos "zlib"
 
   resource "tremor" do
-    url "https://git.xiph.org/tremor.git",
-        :revision => "7c30a66346199f3f09017a09567c6c8a3a0eedc8"
+    url "https://gitlab.xiph.org/xiph/tremor.git",
+        :revision => "b56ffce0c0773ec5ca04c466bc00b1bbcaf65aef"
```
Comment (Bo98) [inline Formula/reminiscence.rb:26]: This one is actually older and contains security vulnerabilities. Seems like they hadn't completely mirrored tremor to GitLab.

Comment (EricFromCanada) [reply Formula/reminiscence.rb]: I found another source for that particular revision.

**Review by SMillerDev — DISMISSED** [review]

File: Formula/reminiscence.rb (hunk 1 above)
Comment (Bo98) [reply Formula/reminiscence.rb]: Looks good. `git.xiph.org` will return eventually so can easily use that again for `tremor` in the future.

File: Formula/gst-validate.rb [hunk 2]
```diff
@@ -23,6 +20,7 @@ class GstValidate < Formula
 
   depends_on "gobject-introspection" => :build
   depends_on "pkg-config" => :build
+  depends_on "python" => :test
```
Comment (Bo98) [inline Formula/gst-validate.rb:23]: Why is this needed? Sound like an issue - we don't want anything using 3.7 anymore.

File: Formula/reminiscence.rb [hunk 3]
```diff
@@ -21,9 +21,10 @@ class Reminiscence < Formula
 
   uses_from_macos "zlib"
 
+  # https://gitlab.xiph.org/xiph/tremor.git is currently missing the latest patches
   resource "tremor" do
-    url "https://git.xiph.org/tremor.git",
-        :revision => "7c30a66346199f3f09017a09567c6c8a3a0eedc8"
+    url "http://sources.buildroot.org/tremor/tremor-7c30a66346199f3f09017a09567c6c8a3a0eedc8.tar.gz"
+    sha256 "ba94cfdf886399c550f76908285bfa9e322f24085de6f1810c2abea565c13a15"
   end
```
Comment (Bo98) [inline Formula/reminiscence.rb:28]: Looks like they've fixed the issue and the commits are there now on GitLab.

File: Formula/gst-validate.rb [hunk 4]
```diff
@@ -49,6 +47,8 @@ def install
       system "make"
       system "make", "install"
     end
+
+    inreplace bin/"gst-validate-launcher", "#!/usr/bin/env python3", "#!#{Formula["python@3.8"].opt_bin}/python3"
```
Comment (Bo98) [inline Formula/gst-validate.rb:51]: It doesn't matter much, but the recommended way is `rewrite_shebang detected_python_shebang, bin/"gst-validate-launcher"` (needs a `include Language::Python::Shebang` at the top of the formula).

Comment (Bo98) [reply Formula/gst-validate.rb:51]: Oh and this one should probably have a revision bump.

**Comment by EricFromCanada** [comment]
```
==> brew install --only-dependencies --include-test pytouhou
==> brew test --verbose pytouhou
==> FAILED
Error: pytouhou is missing test dependencies: mercurial
```
Looks like `brew install --only-dependencies --include-test` doesn't take into account dependencies that come from the URL pointing to a non-Git VCS.

**Comment by Bo98** [comment]
I'm not entirely sure why that's marked as a _test_ dependency. It shouldn't need mercurial during testing. I'll have a look.

**Comment by EricFromCanada** [comment]
Possibly fixed by https://github.com/Homebrew/brew/pull/7633…

**Review by dawidd6 — APPROVED** [review]
